### PR TITLE
A major step towards feature parity with the Ruby SDK

### PR DESCRIPTION
**Pull request by robkinyon**
There's a ton of new features in this one. I think we'll want to cut a new release after this one.

### CODE REVIEW

**Comment by singingfish** [comment]
There's a CI problem.  The patch below fixes it the failure although `-=pod` and `-=cut` may be the real fix:

```

diff --git a/lib/WebService/Braintree/Dispute/Evidence.pm b/lib/WebService/Braintree/Dispute/Evidence.pm
index 330fca5..929abb9 100644
--- a/lib/WebService/Braintree/Dispute/Evidence.pm
+++ b/lib/WebService/Braintree/Dispute/Evidence.pm
@@ -9,2 +9,3 @@ extends "WebService::Braintree::ResultObject";
 =pod
+
 has comment => (is => 'rw');
@@ -14,2 +15,3 @@ has sent_to_processor_at => (is => 'rw');
 has url => (is => 'rw');
+
 =cut

```

File: lib/WebService/Braintree/Gateway.pm [hunk 1]
```diff
@@ -3,108 +3,42 @@ package WebService::Braintree::Gateway;
...
+  subscription => 'Subscription',
+  transaction => 'Transaction',
+  transparent_redirect => 'TransparentRedirect',
+  webhook_notification => 'WebhookNotification',
+  webhook_testing => 'WebhookTesting',
+);
+
+while (my ($method, $gateway) = each %gateways) {
+  my $package = "WebService::Braintree::${gateway}Gateway";
+  eval "use $package"; die $@ if $@;
```
Comment (singingfish) [inline lib/WebService/Braintree/Gateway.pm]: not hugely fond of the string eval here. Moose provides Class::Load::XS so I've made a diff and tested it (see end of review)

**Comment by singingfish** [comment]
this looks pretty solid.  Couple of minor things + changelog + test doc at https://gist.github.com/singingfish/ec67a53dc3af3132b23f70488266f3d3 if you could let me know what you think.  Following that I'll do a release.

**Comment by singingfish** [comment]
Hmm, I nuked the sandbox and tried again, and some sandbox tests were failing for me, although after some effort of manual running I did seem to get most non-deterministic failures passing through the magic of repetition. Main culprits were disbursement and dispute.  Also paypal.  May be the sandbox throwing 500 errors periodically according to my test log

**Comment by robkinyon** [comment]
I've applied the Class::Load patch. Their sandbox is crappy - it randomly 500's, but I don't want to build in retries against 500's because they could be legitimate. I'm about to work on improving the HTTP transport overlay, so maybe I'll add some stuff in there if I find it in the Python, Ruby, and/or JS libraries.

**Comment by singingfish** [comment]
merged from command line, pushed and released to the cpan.

**Pull request closed without merging**